### ISSUE TITLE
Request: RealmConfiguration should have initial data transaction option

### ISSUE THREAD

**Issue by dhiraj** [realm/realm-cocoa#3877]
## Goals
Realm-Java has a very nice method in RealmConfiguration which lets me set up an "initial" data transaction, where I can add all the objects that I need, it is only triggered the first time that I access the realm instance, and never after that.

## Expected Results
Realm-Cocoa should also have this option in RealmConfiguration so that we can set up code based initial data population, instead of having to depend on a separate, bundled Realm database.

**Comment by TimOliver** [comment]
Thanks a lot for the suggestion @dhiraj! I've added this to our backlog so we can review it later. :)

**Comment by jpsim** [comment]
Since `RLMRealm` has an [`isEmpty`](https://realm.io/docs/objc/latest/api/Classes/RLMRealm.html#/c:objc(cs)RLMRealm(py)isEmpty) property, I don't see us adding this any time soon since it's just as easy to use that to accomplish what you'd like.

**Comment by JadenGeller** [comment]
@jpsim I'm not sure `isEmpty` is a total replacement for this feature. Imagine an app that launches populated sample data (by checking `isEmpty`). If the user deletes all the sample data and relaunches the app, it will again be repopulated with sample data.

That said, it's still possible to accomplish the desired effect by inserting some singleton state object after first launch, though this is a bit more tricky for users.

**Comment by jpsim** [comment]
True. There's also the matter of requiring an inter-process lock to avoid race conditions, I just don't think this should be prioritized higher than it is now is all.
